### PR TITLE
Fix java getter / setter name generation

### PR DESCRIPTION
**Pull request by CharlesG-Branch**
Fix for https://github.com/Kotlin/dokka/issues/2310 I have not generated any javadocs with this yet so @IgnatBeresnev if you have any suggestions about where I could add tests for that I can add them in as well.

I couldn't find full documentation from kotlin about how these names map beyond:
- https://kotlinlang.org/docs/java-interop.html#getters-and-setters
- https://kotlinlang.org/docs/java-to-kotlin-interop.html#properties

So to double check the edge cases I copied the TestCase class into our own project using kotlin 1.4.20 and saw what java methods were created.

### CODE REVIEW

**Review by IgnatBeresnev — COMMENTED** [review]
Good job, thanks!

As for javadoc-specific html tests, you can add render-stage tests into the javadoc plugin. Here's a gist to get you started: https://gist.github.com/IgnatBeresnev/434e0e123e813d7c7209b144139e5925

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 1]
```diff
@@ -606,9 +606,17 @@ private class DokkaDescriptorVisitor(
             )
 
         val name = run {
-            val modifier = if (isGetter) "get" else "set"
             val rawName = propertyDescriptor.name.asString()
-            "$modifier${rawName.capitalize()}"
+            // https://kotlinlang.org/docs/java-interop.html#getters-and-setters
+            val specialCaseIs = rawName.startsWith("is")
+                    && rawName.length >= 3
+                    && !rawName[2].isLowerCase()
```
[the remark this answers is not in the document]
Comment (CharlesG-Branch) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:613]: We can't use `isUpperCase` because then it won't handle the edge cases of numbers and underscores. I'll add a comment

File: plugins/base/src/test/kotlin/translators/Issue2310.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package translators
+
+import org.junit.jupiter.api.Assertions.*
+import org.jetbrains.dokka.base.testApi.testRunner.BaseAbstractTest
+import org.junit.jupiter.api.Test
+
+class Issue2310 : BaseAbstractTest() {
```
[the remark this answers is not in the document]
Comment (CharlesG-Branch) [reply plugins/base/src/test/kotlin/translators/Issue2310.kt:7]: Keeping it all together does make it easier to sanity check via copy paste (like I originally did), but maybe that's the less useful case now

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev What should be done with https://github.com/Kotlin/dokka/issues/2017#issuecomment-877153619 ? I might as well knock it out too.

**Comment by IgnatBeresnev** [comment]
That's a good question! I:)  don't mind, but the problem is I don't know what it's used for and how to check it (although I didn't look into that too deep) 

If you manage to find what it actually changes and also correct it, it would be great, but I'm not going to insist :)

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev I see. It looks like it should be an easy fix since it should just inherit the name rather than recreating it, but if we can't test it I won't change it. I'll timebox investigating it.

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev I've added the associated tests. As part of it, I believe I found a bug related to how kotlin link formats (e.g. `[TestCase.issuesFetched]`) are converted into java. I've documented it in the PR.

**Review by IgnatBeresnev — APPROVED** [review]
Left some minor comments, otherwise lgtm, thanks for contributing! :)

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev Done

**Review by vmishenev — APPROVED** [review]

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev When do you think that a release / build with this is going to go out?

**Comment by IgnatBeresnev** [comment]
@CharlesG-Branch dokka 1.6.20 should be out closer to the end of March

 I can make a dev release for you, but you'll have to add an additional maven repository to your build script

**Comment by CharlesG-Branch** [comment]
@IgnatBeresnev That'd be appreciated if it doesn't take much time.
